jack-core/src/rb/schema_rb_parser.rb

```diff
@@ -12,49 +12,118 @@
-            field_defn = FieldDefn.new(
-                field_name,
-                matches[1].to_sym,
-                ordinal,
-                FieldDefn.parse_option_fields(matches[3])
-            )
-
-            model_defn.fields << field_defn
-            ordinal += 1
-          end
-          line = file_lines.shift
-        end
-        models << model_defn
+module FromHash
+  def from_hash(ops)
+    ops.each do |k,v|
+      send("#{k}=",v)
+    end
+    self
+  end
+  def initialize(ops={})
+    from_hash(ops)
+  end
+end
+
+module ActiveRecord
+  class Schema
+    def self.define(ops = {}, &b)
+      $schema = s = new(ops)
+      s.instance_eval(&b)
+      s
+    end
+
+    include FromHash
+    fattr(:indexes) { [] }
+    fattr(:tables) { [] }
+    attr_accessor :version
+
+    def create_table(name, ops = {}, &b)
+      table = Table.new(ops.merge(name: name, schema: self))
+      table.instance_eval(&b)
+      self.tables << table
+    end
+
+    def add_index(table, fields, ops = {})
+      self.indexes << Index.new(ops.merge(table: table, fields: fields))
+    end
+  end
+
+  class Index
+    include FromHash
+    attr_accessor :name, :fields, :unique, :length, :table, :using
+  end
+
+  class Table
+    include FromHash
+    attr_accessor :name, :force, :id, :schema
+    fattr(:columns) { [] }
+
+    def __column(type, name, ops = {})
+      self.columns << Column.new(ops.merge(type: type, name: name))
+    end
+
+    %w(integer text datetime boolean string float binary date decimal varbinary).each do |f|
+      define_method(f) do |*args|
+        self.__column(f, *args)
+      end
+    end
+    def to_model_defn
+      return nil if name == 'schema_info'
+      res = ModelDefn.new(42)
+      res.table_name = name
+      res.model_name = name.singularize.camelize
+      res.fields = columns.each_with_index.map { |x,i| x.to_model_defn(i) }.compact
+      res.migration_number = schema.version.to_s
+      res
+    end
+  end
+
+  class Column
+    include FromHash
+    attr_accessor :type, :name, :limit, :null, :default, :precision, :scale
+
+    def to_model_defn(col_index)
+      f = to_h
+
+      name = f.delete('name')
+      type = f.delete('type').to_sym
+      raise "bad" unless name && type
+      return nil if FORBIDDEN_FIELD_NAMES.include?(f['name'])
+
+      if default.kind_of?(Numeric) && [:float,:decimal].include?(type)
+        f['default'] = default.to_f
+      elsif default.kind_of?(String)
+        f['default'] = '"' + default + '"'
+      end
+
+      FieldDefn.new(
+        name,
+        type,
+        col_index,
+        f.symbolize_keys
+      )
+    end
+
+    def to_h
+      res = {}
+      [:precision, :scale, :limit, :default, :type, :name, :null].each do |f|
+        v = send(f)
+        res[f.to_s] = v if !v.nil?
       end
+      res
     end
-    return models, migration_number
+  end
+end
+
+class SchemaRbParser
+  def self.parse(schema_rb, ignored_tables)
```

Comment:
`ignored_tables` is no longer optional, which breaks the main function (line 131, post your change).

Reply:
This is fixed in https://github.com/LiveRamp/jack/commit/a2617c8e26d8e0f11ae10fe92cc9dde7dff8f1d3.